src/mpid/ch4/netmod/ucx/ucx_impl.h

```diff
@@ -17,7 +17,10 @@
 #include <ucs/type/status.h>
 
 #define MPIDI_UCX_COMM(comm)     ((comm)->dev.ch4.netmod.ucx)
-#define MPIDI_UCX_REQ(req)       ((req)->dev.ch4.netmod.ucx)
+
+/* MPIDI_UCX_REQ return a void pointer that needs be cast into local type */
+#define MPIDI_UCX_REQ(req)       ((void *)&((req)->dev.ch4.netmod.ucx))
```

Comment:
This seems hacky to me. Is there some best practice you could cite for why this is preferred over union?

Reply:
Actually, I see that I have added the union type already (with an error there as well). While my comment/opinion still hold, I don't think I need to insist :) I'll update the code to use union.

Reply:
updated.